src/eu/cqse/check/sample/SampleCheckWithParameter.java

```diff
@@ -0,0 +1,62 @@
+package eu.cqse.check.sample;
+
+import eu.cqse.check.base.EntityCheckBase;
+import eu.cqse.check.framework.core.*;
+import eu.cqse.check.framework.core.option.CheckOption;
+import eu.cqse.check.framework.scanner.ELanguage;
+import eu.cqse.check.framework.shallowparser.framework.ShallowEntity;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.regex.Pattern;
+
+
+/**
+ * This class is meant to demonstrate a custom check with parameter for Teamscale. It creates a new finding for every
+ * method encountered in the system, unless the file containing the code is excluded by the regular expression provided
+ * via the parameter.
+ */
+@Check(name = SampleCheckWithParameter.CHECK_NAME,
+		description = "This is a simple sample check that uses a custom parameter.",
+		groupName = CheckGroupNames.BAD_PRACTICE, defaultEnablement = EFindingEnablement.RED,
+		languages = { ELanguage.JAVA }, parameters = { ECheckParameter.ABSTRACT_SYNTAX_TREE })
+public class SampleCheckWithParameter extends EntityCheckBase {
+	static final String CHECK_NAME = "Sample Check with parameter";
+
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	@CheckOption(name = CHECK_NAME
+			+ " - Filename exclude", description = "Regular expressions to exclude certain files.")
+	private String fileNameExclude = ".*Test.java$";
```

Comment:
Document that this can be left empty if no files should be excluded? This was not clear to me, and it took some time to even understand that this works, because the empty RegEx never matches any (non-empty) uniform path.